pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -111,9 +114,76 @@ func AddToManager(mgr manager.Manager, r *ReconcileClusterPool, concurrentReconc
 		return err
 	}
 
+	// Watch for changes to the hive cluster pool admin RoleBindings
+	if err := c.Watch(
+		&source.Kind{Type: &rbacv1.RoleBinding{}},
+		&handler.EnqueueRequestsFromMapFunc{
+			ToRequests: requestsForRBACResources(r.Client, r.logger),
+		},
+	); err != nil {
+		return err
+	}
+
+	// Watch for changes to the hive-cluster-pool-admin-binding RoleBinding
+	if err := c.Watch(
+		&source.Kind{Type: &rbacv1.RoleBinding{}},
+		&handler.EnqueueRequestsFromMapFunc{
+			ToRequests: requestsForCDRBACResources(r.Client, clusterPoolAdminRoleBindingName, r.logger),
+		},
+	); err != nil {
+		return err
+	}
+
 	return nil
 }
 
+func requestsForCDRBACResources(c client.Client, resourceName string, logger log.FieldLogger) handler.ToRequestsFunc {
+	return func(o handler.MapObject) []reconcile.Request {
+		if o.Meta.GetName() != resourceName {
+			return nil
+		}
+		clusterName := o.Meta.GetNamespace()
```

Comment:
I hope this ClusterDeployment.Name == ClusterDeployment.Namespace assumption holds up over time, it's convenient here.